compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -788,11 +788,27 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                             self.tcx.lang_items().range_inclusive_struct(),
                             self.tcx.lang_items().range_to_inclusive_struct(),
                         ];
-                        if const_def_id != None && ranges.contains(&const_def_id) {
+                        if type_def_id != None && ranges.contains(&type_def_id) {
                             let msg = "constants only support matching by type, \
                                 if you meant to match against a range of values, \
                                 consider using a range pattern like `min ..= max` in the match block";
                             e.note(msg);
```

Comment:
If all of the conditions are met to display the structured suggestion, then this note is redundant. I would only show it if the suggestion isn't.